docs/general/dql/selects.rst

```diff
@@ -740,6 +740,50 @@ an upper-bound::
     Both the ``from`` and ``to`` index values are inclusive.
     Using an index greater than the array size results in an empty array.
 
+For selecting records having values with empty or unassigned arrays, like::
+
+    cr> insert into locations (id, name, position, kind, landmarks) values
+    ... (99, 'Voidspace', 0, 'Milliways', []);
+    INSERT OK, 1 row affected (... sec)
+
+.. Hidden: refresh locations
+
+    cr> refresh table locations;
+    REFRESH OK, 1 row affected (... sec)
+
+a suitable expression to match for both ``NULL`` values and empty arrays, is::
+
+    cr> select count(*) from locations where landmarks is null;
+    +----------+
+    | count(*) |
+    +----------+
+    |       14 |
+    +----------+
+    SELECT 1 row in set (... sec)
+
+.. Hidden: drop record again
+
+    cr> delete from locations where id=99;
+    DELETE OK, 1 row affected  (... sec)
+
+For determining the number of items in an array, you can use the
+:ref:`array_length <scalar-array_length>` scalar function::
+
+    cr> select count(*) from locations where
+    ... array_length(landmarks, 1) = 2;
+    +----------+
+    | count(*) |
+    +----------+
+    |        1 |
+    +----------+
+    SELECT 1 row in set (... sec)
```

Comment:
How about linking to "Array functions" (ref `scalar-arrays`) instead? Seems kinda random to point out one specific function but not mention anything about the others.

Could be a "See also" note

Reply:
That is a much better place, indeed. When looking at the documentation of `array_length` over there [^1], I think the example to demonstrate array length computation on non-empty arrays is already enough. Obviously, I missed that spot and will remove this section altogether.

[^1]: https://crate.io/docs/crate/reference/en/4.8/general/builtins/scalar-functions.html#array-length-anyarray-dimension